### ISSUE TITLE
The dataset variables are only being treated as strings

### ISSUE THREAD

**Issue by romasita** [justinhunt/moodle-filter_generico#6]
When trying to use @@USER:id@@ as a variable to the query, it is being treated as a string.  When adding a var_dump on the $vars variable (created in filter.php), the following output is produced:

array(2) { [0]=> string(1) "3" [1]=> string(8) "ELPRPADV" }

I've added the following code after $vars is initialized:
$query_vars = array();
            for($i = 0; $i < sizeof($vars); $i++) {
                if(is_numeric($vars[$i])) {
                    $query_vars[] = intval($vars[$i]);
                }
                else {
                    $query_vars[] = $vars[$i];
                }
            }

And I also made $query_vars the second argument to the get_records_sql function.  Now a var_dump prints out:
array(2) { [0]=> int(3) [1]=> string(8) "ELPRPADV" }

**Comment by justinhunt** [comment]
Thanks for this. I will look at it, and see if I can reproduce the problem and the solution.

**Comment by romasita** [comment]
Has there been any update to this?  We would like to update the plugin on our production site, and the admin would like to use this fix.

**Comment by justinhunt** [comment]
Hi Amy

Sorry for the delay, I had another fix to come out and I was holding off till that was ready. I added you changes, thanks very much, and I have also released it on Moodle.org.  I was a little worried that a non integer and yet numeric value might come in as a var. Its hard to think of a case, but anyway ...

The release also contains a new field for each template for alternate content to be displayed on the mobile app in place of the regular output.
